### PR TITLE
Added new keyboard Origin65A

### PR DESCRIPTION
**Pull request by HorrorTroll**
## Description

Added new keyboard Origin65A

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/finalkey/origin65a/config.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* Bootmagic key */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
+
```
Comment (waffle87) [inline keyboards/finalkey/origin65a/config.h:24]:
```suggestion
```

File: keyboards/finalkey/origin65a/info.json [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
+    "features": {
+        "bootmagic": true,
+        "mousekey": false,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "lto": true,
```
Comment (waffle87) [inline keyboards/finalkey/origin65a/info.json:25]:
```suggestion
```

File: keyboards/finalkey/origin65a/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                                                            └───┴───┴───┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
+    [_FN3]  = LAYOUT(
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+                _______, _______, _______,                   _______,                   _______, _______, _______, _______,          _______, _______
+            ),
```
Comment (waffle87) [inline keyboards/finalkey/origin65a/keymaps/default/keymap.c:144]:
```suggestion
            )
```

File: keyboards/finalkey/origin65a/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [_BASE] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [_FN1]  = { ENCODER_CCW_CW(RGB_RMOD, RGB_MOD) },
+    [_FN2]  = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [_FN3]  = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```
Comment (waffle87) [inline keyboards/finalkey/origin65a/keymaps/default/keymap.c:152]:
```suggestion
    [_FN1]  = { ENCODER_CCW_CW(RGB_RMOD, RGB_MOD) }
```

File: keyboards/finalkey/origin65a/info.json [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        "rgb_matrix": true
+    },
```
Comment (waffle87) [inline keyboards/finalkey/origin65a/info.json:27]:
```suggestion
    },
    "build": {
        "lto": true
    },
```

File: keyboards/finalkey/origin65a/info.json [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+        "device_version": "0.0.1"
+    },
+    "debounce": 5,
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["F5", "F4", "F1", "F0", "E6", "B1", "B2", "B3", "D4", "D6", "D7", "B4", "B5", "B6", "C7"],
+        "rows": ["D0", "D1", "D2", "D3", "D5"]
+    },
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/finalkey/origin65a/info.json:17]: This uses a Pro Micro? If not, this should be `atmel-dfu`.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/finalkey/origin65a/info.json (hunk 6 above)
Comment (HorrorTroll) [reply keyboards/finalkey/origin65a/info.json:17]: i don't know why, but the PCB come with keyboard was using Caterina bootloader.

Comment (waffle87) [reply keyboards/finalkey/origin65a/info.json:17]: No worries then, as long as you're aware and it's not just some copy pasta.

Comment (HorrorTroll) [reply keyboards/finalkey/origin65a/info.json:17]: Yeah, it kinda annoying. But the bootloader was modify and decrease timeout only 4 seconds, not like original was 8 seconds.

**Review by drashna — APPROVED** [review]

File: keyboards/finalkey/origin65a/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+// entirely and just use numbers.
+
+enum layer_names {
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
+};
+
+// enum layer_keycodes { };
```
Comment (zvecr) [inline keyboards/finalkey/origin65a/keymaps/via/keymap.c:33]:
```suggestion
```

File: keyboards/finalkey/origin65a/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+// enum layer_keycodes { };
```
Comment (zvecr) [inline keyboards/finalkey/origin65a/keymaps/default/keymap.c:31]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also,
* `keyboards/finalkey/origin65a/rules.mk` needs to match <https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/rules.mk>

File: keyboards/finalkey/origin65a/keymaps/via/keymap.c (hunk 7 above)
Comment (HorrorTroll) [reply keyboards/finalkey/origin65a/keymaps/via/keymap.c]: Thanks, I've just resolved on new commit

**Comment by Pangorin** [comment]
@HorrorTroll fix pls

**Pull request closed without merging**